ethcore/sync/src/chain/mod.rs

```diff
@@ -1083,6 +1117,9 @@ impl ChainSync {
 				peer_info.last_sent_transactions.clear()
 			);
 		}
+
+		// reset stats for private transaction packets
+		self.clear_private_stats();
```

Comment:
So it seems that we always re-broadcast after the the block, right? Why can't we use the same logic as for regular transactions?

Also this statements should be guarded by similar ifs as the above code - we don't need to clear it when we import uncle block or we are syncing.